util/git/client.go

```diff
@@ -174,6 +174,10 @@ func NewClientExt(rawRepoURL string, root string, creds Creds, insecure bool, en
 	return client, nil
 }
 
+var (
+	gitClientTimeout = env.ParseDurationFromEnv("ARGOCD_GIT_REQUEST_TIMEOUT", 15, 0, math.MaxInt64)
```

Comment:
Can we add some documentation around the variable? Like add it to the environment variables page?

Reply:
done